intellij-ext/modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt

```diff
@@ -91,61 +119,30 @@ class MirrordConfigDropDown : ComboBoxAction() {
     }
 
-
     companion object {
         var chosenFile: String? = null
-        var configFiles: HashSet<String>? = null
-    }
-}
-
-// Based on virtual file events, we update our configs since indexes are always not up to date
-class MirrordConfigWatcher : AsyncFileListener {
-
-
-    private val adderFileVisitor = object : VirtualFileVisitor<Any?>() {
-        override fun visitFile(file: VirtualFile): Boolean {
-            if (!file.isDirectory && file.path.endsWith("mirrord.json")) {
-                synchronized(this) {
-                    MirrordConfigDropDown.configFiles?.add(file.path)
-                }
-            }
-            return true
-        }
-    }
+        var updateConfigs: Boolean = false
 
-    private val removerFileVisitor = object : VirtualFileVisitor<Any?>() {
-        override fun visitFile(file: VirtualFile): Boolean {
-            if (!file.isDirectory && file.path.endsWith("mirrord.json")) {
-                synchronized(this) {
-                    MirrordConfigDropDown.configFiles?.remove(file.path)
-                }
-            }
-            return true
+        fun matches(path: String, basePath: String?): Boolean {
+            // TODO: we need to also parse toml and yaml
+            return FileSystems.getDefault().getPathMatcher("glob:**/*mirrord.json")
+                .matches(Path.of(path)) && basePath?.let { path.startsWith(it) } ?: true
         }
     }
+}
 
-
+// `configFiles` are updated per VFS changes, this approach helps us avoid querying the index on click
+// and also updates the configFiles in case the selected file is deleted
+class MirrordConfigWatcher : AsyncFileListener {
     override fun prepareChange(events: MutableList<out VFileEvent>): AsyncFileListener.ChangeApplier {
         return object : AsyncFileListener.ChangeApplier {
-            override fun beforeVfsChange() {
-                events.forEach { event ->
-                    when (event) {
-                        // In case of a file the fileVisitor will just check that file,
-                        // but for directories we need to check recursively
-                        is VFileDeleteEvent, is VFileMoveEvent -> {
-                            event.file?.let {
-                                visitChildrenRecursively(it, removerFileVisitor)
-                            }
-                        }
-                    }
-                }
-            }
-
             override fun afterVfsChange() {
                 events.forEach { event ->
                     when (event) {
-                        is VFileCreateEvent, is VFileMoveEvent, is VFileCopyEvent -> {
-                            event.file?.let { file -> visitChildrenRecursively(file, adderFileVisitor) }
+                        is VFileCreateEvent, is VFileDeleteEvent, is VFileMoveEvent, is VFileCopyEvent -> {
+                            synchronized(this) {
+                                MirrordConfigDropDown.updateConfigs = true
```

Comment:
If we have the file in our hands right here, why set the flag to rescan the index and not directly make just the new change (like you did before)?

Reply:
the indexes are always updated anyway, it is not scalable to do that manual work